src/MANYBODY/pair_airebo.cpp

```diff
@@ -3094,72 +3092,54 @@ double PairAIREBO::gSpline(double costh, double Nij, int typei,
 double PairAIREBO::PijSpline(double NijC, double NijH, int typei, int typej,
                              double dN2[2])
 {
-  int x,y,i,done;
-  double Pij,coeffs[16];
-
-  for (i = 0; i < 16; i++) coeffs[i]=0.0;
+  int x,y;
+  double Pij;
 
   x = 0;
   y = 0;
   dN2[0] = 0.0;
   dN2[1] = 0.0;
-  done = 0;
+  Pij = 0.0;
 
-  // if inputs are out of bounds set them back to a point in bounds
+  if (typei == 1) return Pij;
 
-  if (typei == 0 && typej == 0) {
-    if (NijC < pCCdom[0][0]) NijC=pCCdom[0][0];
-    if (NijC > pCCdom[0][1]) NijC=pCCdom[0][1];
-    if (NijH < pCCdom[1][0]) NijH=pCCdom[1][0];
-    if (NijH > pCCdom[1][1]) NijH=pCCdom[1][1];
+  if (typej == 0) {
 
-    if (fabs(NijC-floor(NijC)) < TOL && fabs(NijH-floor(NijH)) < TOL) {
-      Pij = PCCf[(int) NijC][(int) NijH];
-      dN2[0] = PCCdfdx[(int) NijC][(int) NijH];
-      dN2[1] = PCCdfdy[(int) NijC][(int) NijH];
-      done = 1;
-    }
-    if (done == 0) {
-      x = (int) (floor(NijC));
-      y = (int) (floor(NijH));
-      for (i = 0; i<16; i++) coeffs[i] = pCC[x][y][i];
-      Pij = Spbicubic(NijC,NijH,coeffs,dN2);
-    }
-  }
-
-  // if inputs are out of bounds set them back to a point in bounds
+    // if inputs are out of bounds set them back to a point in bounds
 
-   if (typei == 0 && typej == 1){
-     if (NijC < pCHdom[0][0]) NijC=pCHdom[0][0];
-     if (NijC > pCHdom[0][1]) NijC=pCHdom[0][1];
-      if (NijH < pCHdom[1][0]) NijH=pCHdom[1][0];
-      if (NijH > pCHdom[1][1]) NijH=pCHdom[1][1];
+    if (NijC <  pCCdom[0][0]) NijC=pCCdom[0][0];
+    if (NijC >= pCCdom[0][1]) NijC=pCCdom[0][1]-TOL;
+    if (NijH <  pCCdom[1][0]) NijH=pCCdom[1][0];
+    if (NijH >= pCCdom[1][1]) NijH=pCCdom[1][1]-TOL;
+    x = (int) floor(NijC);
```

Comment:
That is fair criticism. I will take another look.

I would like to point out, though, that the spline data in the CH.airebo file is apparently generated in with single precision floating point math, i.e 7-8 valid digits precision and then printed with more digits, as can be seen from numbers like  -0.1640341799999999  or 0.1093561200000001. So the splined/tabulated data is already more noisy than what is caused by the shortcut i am adding here.